### PR TITLE
Overall test improvements

### PR DESCRIPTION
**Pull request by ddreimane**
Overall test improvement of the system.

### CODE REVIEW

File: tests/de/gurkenlabs/litiengine/GameRandomTests.java [hunk 1]
```diff
@@ -53,4 +111,108 @@ public void chooseStringArray_ThrowsOnEmpty() {
...
+        Integer result = this.gameRandom.choose(collection);
+
+        // assert
+        assertNull(result);
+    }
+
+    @Test
+    public void chooseIntegerCollection_empty(){
+        // arrange
+        Collection<Integer> collection = List.of();
```
Comment (Gamebuster19901) [inline tests/de/gurkenlabs/litiengine/GameRandomTests.java:130]: List.of() does not exist in Java 8.

**Review by Gamebuster19901 — CHANGES_REQUESTED** [review]
Please correct the following build error:
```
gamebuster@gamebuster-System-Product-Name:~/Desktop/Modding/LitiEngine$ ./gradlew build

> Task :compileTestJava FAILED
/home/gamebuster/Desktop/Modding/LitiEngine/tests/de/gurkenlabs/litiengine/GameRandomTests.java:130: error: cannot find symbol
        Collection<Integer> collection = List.of();
                                             ^
  symbol:   method of()
  location: interface List
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
6 actionable tasks: 4 executed, 2 up-to-date
```

File: tests/de/gurkenlabs/litiengine/GameRandomTests.java (hunk 1 above)
Comment (niels89) [reply tests/de/gurkenlabs/litiengine/GameRandomTests.java]: @Gamebuster19901 I replaced the List.of() call and checked for other usages (was the only one), should be fixed now.

**Review by nightm4re94 — CHANGES_REQUESTED** [review]
Thank you! With the most recent changes to the PropTests, there is now a merge conflict. Please address that before I can merge.

**Comment by jluech** [comment]
@nightm4re94 added a small cleanup and force pushed to rebase against current master

**Review by nightm4re94 — APPROVED** [review]
I appreciate your efforts, thanks a ton!

**Pull request merged**